packages/lesswrong/components/comments/CommentsListSection.tsx

```diff
@@ -305,6 +237,116 @@ const CommentsListSection = ({
+  const newCommentsSinceDate = highlightDate
+    ? filter(
+      comments,
+      (comment) => new Date(comment.postedAt).getTime() > new Date(highlightDate).getTime(),
+    ).length
+    : 0;
+  const now = useCurrentTime();
+
+  const [anchorEl,setAnchorEl] = useState<HTMLElement|null>(null);
+
+  const handleClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  }
+
+  const handleDateChange = (date: Date) => {
+    setHighlightDate(date)
+    setAnchorEl(null);
+  }
+
+
+  const suggestedHighlightDates = [moment(now).subtract(1, 'day'), moment(now).subtract(1, 'week'), moment(now).subtract(1, 'month'), moment(now).subtract(1, 'year')]
+  const newLimit = commentCount + (loadMoreCount || commentCount)
+  let commentSortNode = (commentCount < totalComments) ?
+    <span>
+      Rendering {commentCount}/{totalComments} comments, sorted by <CommentsViews post={post} setRestoreScrollPos={setRestoreScrollPos} />
+      {loadingMoreComments ? <Loading /> : <a onClick={() => loadMoreComments(newLimit)}> (show more) </a>}
+    </span> :
+    <span>
+      {postGetCommentCountStr(post, totalComments)}, sorted by <CommentsViews post={post} setRestoreScrollPos={setRestoreScrollPos} />
+    </span>
+  if (isFriendlyUI) {
+    commentSortNode = <>Sorted by <CommentsViews post={post} setRestoreScrollPos={setRestoreScrollPos} /></>
+  }
+
+  const contentType = isEAForum && post?.shortform
+    ? "quick takes"
+    : "comments";
+
+  return <CommentsListMeta>
+    <Typography
+      variant="body2"
+      component='span'
+      className={classes.commentSorting}
+    >
+      {commentSortNode}
+    </Typography>
+    {post && !hideDateHighlighting && <Typography
+      variant="body2"
+      component='span'
+      className={classes.clickToHighlightNewSince}
+    >
+      {highlightDate && newCommentsSinceDate>0 && `Highlighting ${newCommentsSinceDate} new ${contentType} since `}
+      {highlightDate && !newCommentsSinceDate && `No new ${contentType} since `}
+      {!highlightDate && `Click to highlight new ${contentType} since: `}
+      <a className={classes.button} onClick={handleClick}>
+        <CalendarDate date={highlightDate || now}/>
+      </a>
+      <Menu
+        anchorEl={anchorEl}
+        open={Boolean(anchorEl)}
+        onClose={handleClose}
+      >
+        {currentUser && post && <LastVisitList
+          postId={post._id}
+          currentUser={currentUser}
+          clickCallback={handleDateChange}/>}
+        <SimpleDivider />
+        {suggestedHighlightDates.map(date => {
+          return <MenuItem key={date.toString()} onClick={() => handleDateChange(date.toDate())}>
+            {date.calendar().toString()}
+          </MenuItem>
+        })}
+      </Menu>
+    </Typography>}
+  </CommentsListMeta>
+}
+
+export default registerComponent("CommentsListSection", CommentsListSection, {
+  areEqual: "auto"
```

Comment:
Was there a part of the deep comparison in the `CommentsListSection` props which triggered (unnecessary) comments list section rerenders?  I guess most client-side changes to the `post` might not be relevant here?

Reply:
(actually it looks like according to `memoizeComponent` that `areEqual: 'auto'` just causes it to memoize the component, which otherwise doesn't happen by default, and it uses whatever the default behavior of React.memo is, which I assume is reference equality?)

Reply:
Right, `areEqual: "auto"` is shallow comparison only.